lib/ansible/modules/windows/win_disk_setup.py

```diff
@@ -0,0 +1,324 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['stableinterface'],
+                    'supported_by': 'core'}
+
+
+DOCUMENTATION = r'''
+---
+module: win_disk_setup
+version_added: '2.4'
+short_description: A windows disk module
+description:
+   - Find a disk on the server and set it up. To find the disk and to setup the disk you have several options.
+options:
+  FindSize:
+      description:
+        - Size of the disk to find
+      required: true
+      default: 5
+      aliases: []
+  FindPartitionStyle:
+      description:
+        - Partition style which should be set on the disk to find
+      required: false
+      default: RAW
+      choices:
+        - RAW
+        - MBR
+        - GPT
+  FindOperationalStatus:
+      description:
+        - Operational Status which should be set on the disk to find
+      required: false
+      default: Offline
+      choices:
+        - Offline
+        - Online
+  SetPartitionStyle:
+      description:
+        - Partition style which should be set on found disk
+      required: false
+      default: GPT
+      choices:
+        - GPT
+        - MBR
+  SetDriveLetter:
+      description:
+        - Drive letter which should be set for the partition on found disk (Protected letters are C and D)
+      required: false
+      default: E
+  SetAllocUnitSize:
+      description:
+        - Allocation Unit size which should be set for the file system on found disk (possible values for file system NTFS 4,8,16,32,64KB; ReFs 64KB)
+      required: false
+      default: 4
+      choices:
+        - 8
+        - 16
+        - 32
+        - 64
+  SetFileSystem:
+      description:
+        - File system which should be set on found disk
+      required: false
+      default: NTFS
+      choices:
+        - NTFS
+        - ReFs
+  SetLabel:
+      description:
+        - File system label which should be set for the file system on found disk
+      required: false
+      default: AdditionalDisk
+  SetLargeFRS:
+      description:
+        - Switch to set Large FRS option for file system on found disk, only for NTFS file system
+      required: false
+      default: false
+      choices:
+        - true
+        - false
+  SetShortNames:
+      description:
+        - Switch to set Short Names option for file system on found disk, only for NTFS file system
+      required: false
+      default: false
+      choices:
+        - true
+        - false
+  SetIntegrityStreams:
+      description:
+        - Switch to set Integrity Streams option for file system on found disk, only for ReFs file system
+      required: false
+      default: false
+      choices:
```

Comment:
So `required: false` should be removed and this replaces `default` and `choices`:
```yaml
type: bool
default: 'no'
```